src/averageAges.js

```diff
@@ -15,11 +15,17 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const persons = people.filter(person => person.sex === 'm');
+
+  const midAgeAll = persons.reduce((sum, man) =>
+    (sum + man.died - man.born), 0) / persons.length;
+
+  const manInCentry = persons.filter(man =>
+    Math.ceil(man.died / 100) === century);
+  const midAgeCentry = manInCentry.reduce((sum, man) =>
+    (sum + man.died - man.born), 0) / manInCentry.length;
+
+  return century ? midAgeCentry : midAgeAll;
```

Comment:
It's better to add a check for century existence at the start of function and delete redundant second reduce method:
```suggestion
  const persons = century
    ? people.filter(person => (
      person.sex === 'm' && Math.ceil(man.died / 100) === century)
    ));
    : people.filter(person => person.sex === 'm');

  const midAgeAll = persons.reduce...

  return midAgeAll;
```